### DISCUSSION TITLE
Are Fast Data Projection indexes all strictly necessary?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#502] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12.4.0

### Description

Hi guys,

I have noticed that the Fast Data creates the following indexes for each projection:
- mia_internal_counter_index
- mia_primary_key_index
- mia_internal_counter_type_index
- mia_state_index

Since they are quite heavy, I am wondering if they are all stricly necessary or some of them can be safely deleted under certain circumstances.

Thanks

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @Lucascanna !

Here I am again with some good news: If you are using the Projection Storer in the ingestion step, the indexes `mia_internal_counter_type_index` and `mia_internal_counter_index` can be removed since the fields `__internal__counter` and `__internal__kafkaInfo` are populated only by the RealTime Updater. 

@albertotessarotto we should update the doc or place a warning in the index section suggesting that those indexes are not needed in case of the Projection Storer.

**Reply by albertotessarotto** [reply]
sure, we start by updating the docs accordingly, then let's track the UI improvement activity that @flower-of-the-bridges suggests

**Reply by lucascanna** [reply]
Thanks @flower-of-the-bridges for the perfect support as usual.

Since these are internal indexes used by the Fast Data, they should be created / deleted automatically by the Console based on which service I am using in my configuration.
